### PR TITLE
MWPW-155057: Lets authors add "starting at" to merch cards

### PR DESCRIPTION
**Pull request by Axelcureno**
Lets authors add "Starting at" to merch card content before the price.
- "Starting at" content can be added using the block "property" with the same value.
- Content is inserted using placeholder {{starting-at}}.

Supported merch card variants:

- Segment
- Product
- Mini compare
- Catalog

Screenshot 2024-08-08 at 1 12 07 PM

Resolves: [MWPW-155057](https://jira.corp.adobe.com/browse/MWPW-155057)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/axel/stories/mwpw-155057?martech=off
- After: https://mwpw-155057--milo--adobecom.hlx.page/drafts/axel/stories/mwpw-155057?martech=off

### CODE REVIEW

**Comment by narcis-radu** [comment]
Hey @Axelcureno - I'm not able to see the "starting at" text in your demo page https://mwpw-155057--milo--adobecom.hlx.page/drafts/axel/stories/mwpw-155057?martech=off

**Comment by robert-bogos** [comment]
@Axelcureno Should we add some unit tests for this?

**Comment by Axelcureno** [comment]
> Hey @Axelcureno - I'm not able to see the "starting at" text in your demo page https://mwpw-155057--milo--adobecom.hlx.page/drafts/axel/stories/mwpw-155057?martech=off

@narcis-radu that's strange, I still see the change. Maybe try clearing cache?

**Review by mirafedas — APPROVED** [review]

**Comment by mirafedas** [comment]
@Axelcureno I guess we probably should update the authoring documentation after this change is live

File: libs/blocks/merch-card/merch-card.js [hunk 1]
```diff
@@ -403,6 +402,19 @@ const updateBigPrices = (merchCard) => {
   });
 };
 
+const addStartingAt = async (styles, merchCard) => {
+  if (styles.includes('starting-at')) {
+    const { replaceKey } = await import('../../features/placeholders.js');
+    await replaceKey('starting-at', getConfig()).then((key) => {
+      const startingAt = createTag('div', { slot: 'starting-at' }, key);
+      const prices = merchCard.querySelectorAll('span[is="inline-price"]');
```
Comment (npeltier) [inline libs/blocks/merch-card/merch-card.js:410]: please rather use `querySelector` if you are just going to use the first one

**Review by npeltier — APPROVED** [review]
nit request

**Comment by afmicka** [comment]
@Axelcureno your changes are adding &nbsp; within the price. Could you please take a look? 
Could you update the branch with the stage latest and make sure you don't have stale block so that we can validate the PR?

Screenshot 2024-08-14 at 13 36 51

**Comment by Axelcureno** [comment]
> @Axelcureno your changes are adding/exposing `&nbsp;` within the price. Could you please take a look? cc: @Roycethan
> 
> Could you also update the branch with the stage latest and make sure you don't have stale block so that we can validate the PR? This issue has been seen in the recent Stage to Main PRs.
> 
> https://main--cc--adobecom.hlx.live/pl/creativecloud?milolibs=mwpw-155057
> 
> Screenshot 2024-08-14 at 13 36 51

@afmicka It wasn't my changes that were adding the `&nbsp;` but I have updated my branch with latest stage and the characters are now gone.

**Review by Roycethan — APPROVED** [review]

**Review by robert-bogos — APPROVED** [review]

**Pull request merged**